File: src/main/java/com/example/project_test/Entities/Chambre.java
package com.example.project_test.Entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.List;
import java.util.Set;
@Getter
@Setter

@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Chambre implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long idChambre;
    long numeroChambre;
    @Enumerated(EnumType.STRING)
    TypeChambre typeC;
    @ManyToOne
    Bloc bloc;
    @OneToMany
    private Set<Reservation> reservations;


}

File: src/app/model/Chambre.ts
import { Bloc } from "./bloc";
import { TypeChambre } from "./typeChambre";

export class Chambre{
  idChambre!:number;
  numeroChambre!:number;
  typeChambre?:TypeChambre;
  bloc!:Bloc;
  // dateDebut!:Date;
  // dateFin!:Date;

}
